### ISSUE TITLE
Set sensible default value for `chroot`

### ISSUE THREAD

**Issue by ThomasLandauer** [nucleos/NucleosDompdfBundle#181]
## Feature Request

Since 0.8.6 dompdf requires a `chroot` setting for loading assets. From the fact that around 10 (!) issues were opened as a result ( see https://github.com/dompdf/dompdf/issues/2229 for an overview), I'm concluding that this is causing people some trouble ;-)

https://github.com/dompdf/dompdf/wiki/Usage#options recommends setting an "absolute path".

So I'm suggesting to include a default value for this, which makes sense for Symfony. Maybe `%kernel.project_dir%/public`?

**Comment by core23** [comment]
Can you provide a PR @ThomasLandauer ?

**Comment by ThomasLandauer** [comment]
I would say that this should go into `config/packages/nucleos_dompdf.yaml`:
```yaml
nucleos_dompdf:
    defaults:
        chroot: '%kernel.project_dir%/public'
```

2 questions:
1. `chroot` is to prevent sensitive files from being exposed through dompdf, see https://github.com/dompdf/dompdf/tree/v1.0.1#resource-reference-requirements
    Setting this to `public` might lead to Symfony's `index.php` https://github.com/symfony/recipes/blob/master/symfony/framework-bundle/5.2/public/index.php getting exposed.
    I think this is OK, since this file doesn't contain any sensitive information, and the setting can be adjusted by the user anyway. What do you think?
2. Am I right that there are two ways to create the config file upon installation: Creating a Symfony Flex recipe or using [endroid/installer](https://github.com/endroid/installer)? Which do you prefer?

**Comment by core23** [comment]
>     1. Setting this to `public` might lead to Symfony's `index.php` https://github.com/symfony/recipes/blob/master/symfony/framework-bundle/5.2/public/index.php getting exposed.

Maybe you could use `'%kernel.project_dir%/public/assets` in the example to reduce the risk of exposing any sensitive file.

> [...] using [endroid/installer](https://github.com/endroid/installer)? Which do you prefer?

Never used the installer. The endroid lib did not exist when I started this bundle

**Comment by ThomasLandauer** [comment]
`'%kernel.project_dir%/public/assets`: I was thinking about this too. But I'm not sure how common it is to put everything into `assets`. Is this an "official" Symfony recommendation? Couldn't find any reference yesterday...

Flex vs. endroid/installer: I haven't used either, so I can't recommend anything. I guess it's down to the question: External reviewing process (=Flex) vs. additional dependency (=endroid). You decide :-)

**Comment by core23** [comment]
> `'%kernel.project_dir%/public/assets`: I was thinking about this too. But I'm not sure how common it is to put everything into `assets`. Is this an "official" Symfony recommendation? Couldn't find any reference yesterday...

AFAIK there is no best practice for non-script assets. Using `public/assets` is okay for me.

> Flex vs. endroid/installer: I haven't used either, so I can't recommend anything. I guess it's down to the question: External reviewing process (=Flex) vs. additional dependency (=endroid). You decide :-)

You should leave this out as no-one knows anything about this stuff ;)

**Comment by ThomasLandauer** [comment]
> You should leave this out as no-one knows anything about this stuff ;)

Well, what's your plan then to get this file to the user?

**Comment by core23** [comment]
Either create a new recipe for symfony flex: https://github.com/symfony/recipes-contrib or add a short note to the README.

**Comment by ThomasLandauer** [comment]
There you go :-) https://github.com/symfony/recipes-contrib/pull/1075

**Comment by ThomasLandauer** [comment]
TODO: After this gets merged, the following can be removed from README, right?:
> Then, enable the bundle by adding it to the list of registered bundles in config/bundles.php file of your project:

**Comment by core23** [comment]
> TODO: After this gets merged, the following can be removed from README, right?:

Flex is an optional tool. This project should be useable without symfony flex. So a short note could help

**Comment by core23** [comment]
Fixed with #253
